dlme_airflow/drivers/iiif_json.py

```diff
@@ -21,58 +21,63 @@ def __init__(self, collection_url, dtype=None, metadata=None):
         self.record_limit = self.metadata.get("record_limit")
 
     def _open_collection(self):
-        collection_result = requests.get(self.collection_url)
-        for manifest in collection_result.json().get("manifests", []):
+        collection_result = requests.get(self.collection_url).json()
+        for manifest in collection_result.get("manifests", []):
             self._manifest_urls.append(manifest.get("@id"))
 
-    def _open_manifest(self, manifest_url: str):
-        manifest_result = requests.get(manifest_url)
-        manifest_detail = manifest_result.json()
-        record = self._construct_fields(manifest_detail)
+    def _open_manifest(self, manifest_url: str) -> dict:
+        manifest_result = requests.get(manifest_url).json()
+        record = self._extract_specified_fields(manifest_result)
         # Handles metadata in IIIF manifest
-        record.update(self._from_metadata(manifest_detail.get("metadata", [])))
+        record.update(
+            self._extract_manifest_metadata(manifest_result.get("metadata", []))
+        )
         return record
 
-    def _construct_fields(self, manifest: dict) -> dict:
+    def _extract_specified_fields(self, iiif_manifest: dict) -> dict:
         output = {}
         for name, info in self.metadata.get("fields").items():
             expression = self._path_expressions.get(name)
-            result = [match.value for match in expression.find(manifest)]
-            if len(result) < 1:
+            result = [match.value for match in expression.find(iiif_manifest)]
+            if (
+                len(result) < 1
+            ):  # the JSONPath expression didn't find anything in the manifest
                 if info.get("optional") is True:
-                    # Skip and continue
-                    continue
+                    logging.debug(
+                        f"{iiif_manifest.get('@id')} missing optional field: '{name}'; searched path: '{expression}'"
+                    )
                 else:
-                    logging.warn(f"{manifest.get('@id')} missing {name}")
+                    logging.warning(
+                        f"{iiif_manifest.get('@id')} missing required field: '{name}'; searched path: '{expression}'"
+                    )
             else:
-                if len(result) == 1:
+                if (
+                    len(result) == 1
+                ):  # the JSONPath expression found exactly one result in the manifest
                     output[name] = result[0].strip()
-                else:
+                else:  # the JSONPath expression found exactly one result in the manifest
```

Comment:
when pairing this morning @aaron-collier and i discussed simplifying this logic and just keeping the first defined value, and logging a warning message if multiple values were found.  but after experimenting with an actual IIIF manifest from the configs, and some test code, i've come to think i was mis-understanding this logic and how it might try to accumulate fields.  i think the multiple results code path is actually legit IIIF and a real use case for us (see the `iiif_format` field test in `tests/drivers/test_iiif_json.py`, slightly simplified from a real IIIF manifest that the auc config points to).  so i tried to add clarifying comments here, and test cases that exercise this and have comments that tie the tests to this code.

codeclimate is complaining that this method is too complex, and it probably could use simplification, but i figured it might be better to try getting this merged to start, then refining in a follow-on PR.